### ISSUE TITLE
MongoDB Field filters variable in native query with nested fields not working

### ISSUE THREAD

**Issue by sytuacmdyh** [metabase/metabase#11597]
**Describe the bug**
When I use Field filters variable, and `Field to map to` of the variable is a nested field.

**Logs**
```
12-27 11:38:32 WARN middleware.process-userland-query :: Query failure {:status :failed,
 :class java.lang.Exception,
 :error "Error: Field 887 is not present in the Query Processor Store.",
 :stacktrace
 ("--> query_processor.store$fn__37801$field__37806$fn__37807.invoke(store.clj:236)"
  "query_processor.store$fn__37801$field__37806.invoke(store.clj:232)"
  "driver.mongo.query_processor$field__GT_name.invokeStatic(query_processor.clj:117)"
  "driver.mongo.query_processor$field__GT_name.invoke(query_processor.clj:113)"
  "driver.mongo.parameters$field__GT_name.invokeStatic(parameters.clj:35)"
  "driver.mongo.parameters$field__GT_name.invoke(parameters.clj:34)"
  "driver.mongo.parameters$substitute_field_filter.invokeStatic(parameters.clj:58)"
  "driver.mongo.parameters$substitute_field_filter.invoke(parameters.clj:56)"
  "driver.mongo.parameters$substitute_param.invokeStatic(parameters.clj:74)"
  "driver.mongo.parameters$substitute_param.invoke(parameters.clj:61)"
  "driver.mongo.parameters$substitute_STAR_$fn__1864.invoke(parameters.clj:101)"
  "driver.mongo.parameters$substitute_STAR_.invokeStatic(parameters.clj:94)"
  "driver.mongo.parameters$substitute_STAR_.invoke(parameters.clj:91)"
  "driver.mongo.parameters$substitute.invokeStatic(parameters.clj:113)"
  "driver.mongo.parameters$substitute.invoke(parameters.clj:112)"
  "driver.mongo.parameters$parse_and_substitute.invokeStatic(parameters.clj:123)"
  "driver.mongo.parameters$parse_and_substitute.invoke(parameters.clj:120)"
  "driver.mongo.parameters$substitute_native_parameters.invokeStatic(parameters.clj:131)"
  "driver.mongo.parameters$substitute_native_parameters.invoke(parameters.clj:127)"
  "driver.mongo$eval2184$fn__2185.invoke(mongo.clj:219)"
  "query_processor.middleware.parameters.native$expand_inner.invokeStatic(native.clj:39)"
  "query_processor.middleware.parameters.native$expand_inner.invoke(native.clj:30)"
  "query_processor.middleware.parameters$expand_one.invokeStatic(parameters.clj:50)"
  "query_processor.middleware.parameters$expand_one.invoke(parameters.clj:41)"
  "query_processor.middleware.parameters$expand_all$replace_42673__42674.invoke(parameters.clj:59)"
  "mbql.util.match$replace_in_collection$iter__24166__24170$fn__24171.invoke(match.clj:132)"
  "mbql.util.match$replace_in_collection.invokeStatic(match.clj:131)"
  "mbql.util.match$replace_in_collection.invoke(match.clj:126)"
  "query_processor.middleware.parameters$expand_all$replace_42673__42674.invoke(parameters.clj:59)"
  "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:59)"
  "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:53)"
  "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:56)"
  "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:53)"
  "query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:77)"
  "query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:73)"
  "query_processor.middleware.parameters$fn__42689$substitute_parameters_STAR___42694$fn__42695.invoke(parameters.clj:82)"
  "query_processor.middleware.parameters$fn__42689$substitute_parameters_STAR___42694.invoke(parameters.clj:79)"
  "driver.mongo$eval2008$fn__2009$fn__2011$f__702__auto____2013.invoke(mongo.clj:86)"
  "driver.mongo.util$_with_mongo_connection$fn__693.invoke(util.clj:205)"
  "util.ssh$with_ssh_tunnel_STAR_.invokeStatic(ssh.clj:106)"
  "util.ssh$with_ssh_tunnel_STAR_.invoke(ssh.clj:91)"
  "driver.mongo.util$_with_mongo_connection.invokeStatic(util.clj:199)"
  "driver.mongo.util$_with_mongo_connection.invoke(util.clj:194)"
  "driver.mongo$eval2008$fn__2009$fn__2011.invoke(mongo.clj:85)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__41572.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__43018.invoke(resolve_driver.clj:22)"
  "query_processor.middleware.store$initialize_store$fn__46148$fn__46149.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__46148.invoke(store.clj:10)"
  "query_processor.middleware.async$async__GT_sync$fn__37361.invoke(async.clj:23)"
  "query_processor.middleware.async_wait$runnable$fn__40614.invoke(async_wait.clj:89)"),
 :query
 {:native
  {:template-tags
   {:product {:id "ff3a0d86-c966-d4a8-c26d-c6abb606b15a", :name "product", :display-name "product", :type "dimension", :dimension ["field-id" 937], :required false, :widget-type "category"}},
   :query "[{ $match: {{product}} }]",
   :collection "online_sheetData_5dea2a03051d09045b25d065"},
  :type "native",
  :parameters [{:type "category", :target ["dimension" ["template-tag" "product"]], :value ["2号整机"]}],
  :async? true,
  :middleware {:add-default-userland-constraints? true, :userland-query? true},
  :info
  {:executed-by 8,
   :context :ad-hoc,
   :card-id nil,
   :nested? false,
   :query-hash [-107, -71, -85, 35, -98, -49, 74, 100, -102, -98, -70, -115, 90, 115, 0, 16, -15, 44, -9, -62, -52, 86, -94, 73, -31, 88, -29, -34, 11, 85, -101, 111]},
  :constraints {:max-results 10000, :max-results-bare-rows 2000}}}
```

**To Reproduce**
1. create a native query, and select a table that contains a nested field. here is a example of a collection contains documets like
```
    {
        "a": {
            "b": "1"
        }
    },
    {
        "a": {
            "b": "2"
        }
    }
```
2. write mongo query
```
[{ $match: {{foo}} }]
```
and set attr of foo `Variable type=b(the colume defined above)`, `Filter widget type=Category`
3. run query

**Screenshots**
![image](https://user-images.githubusercontent.com/13233969/71500237-0b235f00-289f-11ea-8e9f-2299074a2167.png)

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.
```
{
  "browser-info": {
    "language": "en",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_202-b08",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_202",
    "java.vm.name": "Java HotSpot(TM) 64-Bit Server VM",
    "java.vm.version": "25.202-b08",
    "os.name": "Mac OS X",
    "os.version": "10.14.6",
    "user.language": "en",
    "user.timezone": "Asia/Shanghai"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "mongo"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "run-mode": "prod",
    "version": {
      "tag": "v0.34.0",
      "date": "2019-12-10",
      "branch": "release-0.34.x",
      "hash": "a6bcf1a"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Not so serious and I made the fix locally

**Additional context**
My solution:
change `qp.store/field` to `Field` in line 117 of metabase/driver/mongo/query_processor.clj
branch release-0.34.x (13352156f8a433d8b35727b9883c48f40de0f479, current the latest)

**Comment by sowmya-mani** [comment]
Thanks, awaiting an official release of the fix. Have been facing the same issue too - https://discourse.metabase.com/t/metabase-mongodb-date-field-filter-variable-error-field-1-521-is-not-present-in-the-query-processor-store/8629

**Comment by camsaul** [comment]
Fixed by #11782
